src/js/components/SidebarFilter.js

```diff
@@ -136,34 +138,43 @@ class SidebarFilter extends mixin(QueryParamsMixin) {
   }
 
   getHealthCheckboxes() {
-    let {filterLabels, filterValues} = this.props;
+    let {filterLabels, filterType, filterValues} = this.props;
     let {selectedNodes} = this.state;
 
-    return Object.keys(filterValues).map(filterValue => {
-      let value = filterValues[filterValue].toString();
-      let checked = selectedNodes.indexOf(value) > -1;
-
-      let labelClassSet = classNames(
-        'side-list-item form-row-element form-element-checkbox inverse',
-        'row row-flex flex-align-items-center vertical-center flush clickable',
-        {
-          'filter-active': this.getCountByValue(filterValue) > 0,
-          'filter-checked': checked
-        }
-      );
-
-      return {
-        checked,
-        checkboxLabelClass: 'form-element-checkbox-label flex-grow',
-        value: checked,
-        fieldType: 'checkbox',
-        formElementClass: 'form-row-element checkbox flush',
-        name: filterValue,
-        label:
-          this.getFormLabel(filterLabels[filterValue], filterValue),
-        labelClass: labelClassSet
-      };
-    });
+    return Object.keys(filterLabels)
+      .filter((filterLabel) => {
+        let filterValue = filterValues[filterLabel];
+
+        return filterValue != null &&
+          !(filterType === ServiceFilterTypes.STATUS &&
+          filterValue === ServiceStatusTypes.NA &&
+          this.getCountByValue(filterValue) === 0);
+      })
+      .map((filterLabel) => {
+        let value = filterValues[filterLabel];
+        let checked = selectedNodes.indexOf(value.toString()) > -1;
+
+        let labelClassSet = classNames(
+          'side-list-item form-row-element form-element-checkbox inverse',
+          'row row-flex flex-align-items-center vertical-center flush clickable',
+          {
+            'filter-active': this.getCountByValue(filterLabel) > 0,
```

Comment:
Yes the function naming is a little bit confusing, but it is working correctly here.
`filterLabel` is resolving to the actual value inside of the function.

```JavaScript
  getCountByValue(filterValue) {
    let {props} = this;
    let count = props.countByValue[props.filterValues[filterValue]];

    if (count == null) {
      return 0;
    }

    return count;
  }
```

`props.fitlerValues` contains this e.g.
```JavaScript
Object {UNHEALTHY: 0, HEALTHY: 1, IDLE: 2, NA: 3}
```
Where the `filterLabel` is this:
`"IDLE"`